lock-api/src/test/java/com/palantir/lock/HeldLocksTokenTest.java

```diff
@@ -0,0 +1,31 @@
+package com.palantir.lock;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableSortedMap;
+
+public class HeldLocksTokenTest {
+	@Test
+	public void testSerde() throws IOException {
```

Comment:
I'm a fan of immutables.  I think it is okay for us to have api changes still (deferring to @tmgordeeva), but it may be possible to make this change without breaking the api.  

If you have to break the api (and we are going forward with that plan) it would be nice to tests the waters in pgdev (as the most complicated consumer of atlas) to make sure that the api change can be consumed well.  The logistics here are little bit annoying, but it is mostly made easy by using atlashelper.sh in pgdev (let's talk offline for more on this).